irohad/consensus/yac/impl/cluster_order.cpp

```diff
@@ -11,16 +11,30 @@ namespace iroha {
 
       boost::optional<ClusterOrdering> ClusterOrdering::create(
           const std::vector<std::shared_ptr<shared_model::interface::Peer>>
-              &order) {
+              &order,
+          std::vector<size_t> const &peer_positions) {
         if (order.empty()) {
           return boost::none;
         }
-        return ClusterOrdering(order);
+        return ClusterOrdering(order, peer_positions);
       }
 
       ClusterOrdering::ClusterOrdering(
-          std::vector<std::shared_ptr<shared_model::interface::Peer>> order)
-          : order_(std::move(order)) {}
+          std::vector<std::shared_ptr<shared_model::interface::Peer>> const
+              &order,
+          std::vector<size_t> const &peer_positions) {
+        order_.reserve(order.size());
+        if (!peer_positions.empty()) {
+          BOOST_ASSERT_MSG(peer_positions.size() == order.size(), 
+            "Peer positions must be the same size to define ordering.");
+          
+          for (auto const &i : peer_positions) {
+            order_.emplace_back(order[i]);
+          }
+        } else {
+          order_ = order;
```

Comment:
this seems a completely different piece of logic, not related to the method signature. If we do need such a constructor, let's make it a standalone with single parameter (peers vector &&). otherwise let's remove this branch. anyway, IMHO the size of the constructed order in this method should be equal to the size of `peer_positions` vector. and this should be documented in the docstrig.

Reply:
It should be better to revert on my previous commit with 2 ctors to keep caching posibility and code clearance. We can not use && here, because of no rval usage in YacGateImpl::vote.